src/bank_branch.c

```diff
@@ -0,0 +1,198 @@
+#include "bank_branch.h"
+
+static char *_build_msg(const char *fmt, ...) {
+  va_list args;
+  va_start(args, fmt);
+
+  int size = vsnprintf(NULL, 0, fmt, args);
+  if (size < 0) {
+    return NULL;
+  }
+  va_end(args);
+
+  char *str = malloc(size + 1);
+  va_start(args, fmt);
+
+  vsnprintf(str, size + 1, fmt, args);
+  va_end(args);
+
+  return str;
+}
+
+static void _append_balance_state(BalanceHistory *history, BalanceState state) {
+  uint8_t len = history->s_history_len;
+  if (len > 0) {
+    BalanceState prev = history->s_history[len - 1];
+    timestamp_t elapsed = state.s_time - prev.s_time - 1;
+    for (int i = 0; i < elapsed; i++) {
+      history->s_history[len + i] = prev;
+    }
+    len += elapsed;
+  }
+  history->s_history[len] = state;
+  history->s_history_len++;
```

Comment:
Please, don't forget to add elapsed time here.

Reply:
fixed